File: 53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int tmp=0,max=INT_MIN;
        for(int i=0;i<nums.size();i++){
            tmp+=nums[i];
            if(tmp>max) max=tmp;
            if(tmp<0)   tmp=0;
        }
        return max;
    }
};

File: 53. Maximum Subarray.go
func maxSubArray(nums []int) int {
	tmp := 0
	max := -10000
	for _, i := range nums {
		tmp += i
		if tmp > max {
			max = tmp
		}
		if tmp < 0 {
			tmp = 0
		}
	}
    return max
}